go/src/staleFlagCleaner.go

```diff
@@ -0,0 +1,624 @@
+	valOfExpr := isBot
+	delCallExpr := false
+	doNotDelLiterals := false
+	for ind := 0; ind < len((*names)); ind++ {
+		//check if name is in valueName
+		valOfExpr = sfc.evaluateExprNode((*values)[ind])
+		delCallExpr = false
+
+		doNotDelLiterals = sfc.checkForBoolLiterals((*values)[ind])
+
+		// Need to see what to do with this thing
+		sfc.updateValueMapPost((*names)[ind], (*values)[ind])
+
+		if (valOfExpr != isBot && !doNotDelLiterals) || delCallExpr {
+			var trueIdent dst.Ident
+			if valOfExpr == isTrue {
+				trueIdent.Name = strTrue
+			} else {
+				trueIdent.Name = strFalse
+			}
+			(*values)[ind] = dst.Expr(&trueIdent)
+			if len((*names)) <= 1 {
+				return true
+			}
+		}
+	}
+	if len((*names)) == 0 {
+		return true
+	}
+	return false
+}
+
+// This is the post function of the dstutil.Apply
+/*
+From documentation:
+If post is not nil, and a prior call of pre didn't return false, post is
+called for each node after its children are traversed (post-order). If
+post returns false, traversal is terminated and Apply returns immediately.
+
+Working:
+This wiil be used to delete the nodes and storing the values of expressions.
+*/
+func (sfc *staleFlagCleaner) post(n *dstutil.Cursor) bool {
+	switch d := n.Node().(type) {
+	case *dst.ValueSpec:
+		//need for loop here to delete the specific varible in
+		delVal := sfc.DelValStmt(&d.Names, &d.Values)
+		if delVal {
+			n.Delete()
+		}
+	//GenDecl
+	case *dst.GenDecl:
+		if len(d.Specs) == 0 {
+			switch n.Parent().(type) {
+			case *dst.DeclStmt:
+				//wait Decl stmt
+			default:
+				n.Delete()
+			}
+		}
+	//declstmt
+	case *dst.DeclStmt:
+		if gd := d.Decl.(*dst.GenDecl); gd.Tok == token.VAR {
+			if len(gd.Specs) == 0 {
+				n.Delete()
+			}
+		}
+
+	//Assignment statements
+	case *dst.AssignStmt:
+		//then delete from slice
+		sfc.DelAssStmt(&d.Lhs, &d.Rhs)
+
+	//If cond statements
+	case *dst.IfStmt:
+		// Init of Ifstmt will get handled in assignStmt
+		// Now evaluate the cond. Cond is already refactored due to pre-order traversal.
+		cond := sfc.evaluateExprNode(d.Cond)
+		if cond != isBot {
+			if cond == isFalse {
+				if d.Else != nil {
+					for _, a := range d.Else.(*dst.BlockStmt).List {
+						n.InsertBefore(a)
+					}
+				}
+			} else {
+				for _, a := range d.Body.List {
+					n.InsertBefore(a)
+				}
+			}
+			n.Delete()
+		}
+	//Switch statements
+	case *dst.SwitchStmt:
+		// Init of switchstmt will get handled in assignStmt
+		if d.Tag != nil {
+			valExpr := sfc.evaluateExprNode(d.Tag)
+			if valExpr != isBot {
+				notfoundCase := false
+				for _, exprstmt := range d.Body.List {
+					if exprstmt.(*dst.CaseClause).List != nil { //nil means default case
+						valCaseClause := sfc.evaluateExprNode(exprstmt.(*dst.CaseClause).List[0])
+						if valCaseClause == valExpr {
+							//this has to go out as switch is going to delete
+							for _, bodyEle := range exprstmt.(*dst.CaseClause).Body {
```

Comment:
Is `exprstmt` always of the `CaseClause` type? If so, then why isn't it of the `CaseClause` type in the first place? Do we need a type check here instead of type assertion?